### PR TITLE
Create My profile page and user update function

### PR DESCRIPTION
**Pull request by Briantam0422**
**1. My profile page**
image

**2. handling user update**

**3. Update user password**
image

**4. Modal component - demo code**
image

### CODE REVIEW

**Comment by ialej001** [comment]
Great work so far, Brian. Is there a way for the user to leave the profile page? Maybe we should add a button on the top right corner (like an X with a circular border) that would reroute the user back to the `/dash` route.

**Review by ialej001 — CHANGES_REQUESTED** [review]
Great job Brian!

**Comment by Briantam0422** [comment]
@ialej001 I have added a back button and changed the label and text you've mentioned. Thanks!

File: src/server/api/routers/user.ts [hunk 1]
```diff
@@ -1,9 +1,80 @@
+import { hash } from "~/utils/bcrypt";
+import { z } from "zod";
 import {
   createTRPCRouter,
+  protectedProcedure,
 } from "~/server/api/trpc";
+import { TRPCError } from "@trpc/server";
+import toast from "react-hot-toast";
```
Comment (mpark4656) [inline src/server/api/routers/user.ts:8]: Is this toast import needed in user.ts?

Comment (Briantam0422) [reply src/server/api/routers/user.ts]: removed!

**Comment by Briantam0422** [comment]
I also added an email update function that allows user to change their email and they need to verify the new email address.
image
image
image

**Review by ialej001 — APPROVED** [review]

**Review by mpark4656 — APPROVED** [review]

**Pull request merged**